packages/mgt-chat/src/statefulClient/StatefulGraphChatClient.ts

```diff
@@ -373,15 +423,13 @@ class StatefulGraphChatClient implements StatefulClient<GraphChatClient> {
     this.notifyStateChange((draft: GraphChatClient) => {
       draft.status = 'loading messages';
     });
-    if (this._chatId) {
+    try {
       this._chat = await loadChat(this.graph, this._chatId);
-      const messages: MessageCollection = await loadChatThread(this.graph, this._chatId, this._messagesPerCall);
-      await this.writeMessagesToState(messages);
-    } else {
-      this.notifyStateChange((draft: GraphChatClient) => {
-        draft.status = 'no messages';
-      });
+    } catch (error) {
```

Comment:
I may well be wrong here, but I thought that when a thrown error is unhandled inside an async function it returns to the calling function as a rejected promise containing the error value.

TLDR; if loadChat fails is there any difference to the calling code between this and the version without the try/catch?

Perhaps adding a console.error here might be helpful?

Reply:
If `loadChat` fails in the version without `try/catch`, subscription to notifications happens. The same thing will happen if you make the calls in `Promise.all([loadChatData(), subscribe()])` despite the final value being a "failed" resolution. I figured that subscribing to notifications for messages that failed to load is unnecessary. I'm adding a console.error for it. This will happen in an account switch before the existing selected chat is changed.

Reply:
Great, thanks for the extra context, and yes, I agree with your assessment about subscriptions on chats that failed to load, they are unnecessary.